tests/serverpod_test_server/test_integration/security_context_test.dart

```diff
@@ -0,0 +1,292 @@
+import 'dart:io';
+
+import 'package:serverpod/serverpod.dart';
+import 'package:serverpod_test_server/src/web/routes/root.dart';
+
+import 'package:serverpod_test_server/test_util/test_serverpod.dart';
+import 'package:test/test.dart';
+
+void main() {
+  const certificatePath = 'test_integration/ssl/test_certificate.pem';
+  const keyPath = 'test_integration/ssl/test_key.pem';
+
+  group('Given a api server', () {
+    group('with a security context', () {
+      late Serverpod serverpod;
+
+      setUpAll(() async {
+        serverpod = IntegrationTestServer.create(
+          apiServerSecurityContext: SecurityContext()
+            ..useCertificateChain(certificatePath)
+            ..usePrivateKey(keyPath),
+        );
+
+        await serverpod.start();
+      });
+
+      tearDownAll(() async {
+        await serverpod.shutdown(exitProcess: false);
+      });
+
+      test('then server is running over https', () {
+        expect(serverpod.server.securityContext, isNotNull);
+      });
+
+      test('then server can be accessed over https', () async {
+        var client = HttpClient();
+        client.badCertificateCallback = (cert, host, port) => true;
```

Comment:
Suggestion: Considering that you are loading real certificates and keys I think it would make more sense to test the validation too and not just accept any certificate given.

E.g.

```dart
final context = SecurityContext();

context.setTrustedCertificates(...)
HttpClient(context: context);
```